### ISSUE TITLE
degree9.boot_semver.impl$version_impl$fn__1053 cannot be cast to java.lang.CharSequence

### ISSUE THREAD

**Issue by jgrzebyta** [degree9/boot-semver#59]
Affected versions: *1.5.0*, *1.5.0-SNAPSHOT*

In the mock project command `boot version` causes `degree9.boot_semver.impl$version_impl$fn__1053 cannot be cast to java.lang.CharSequence` exception.

Full stack trace:
```
Initial Project Version...: 0.0.4
                              java.lang.Thread.run                      Thread.java:  745
java.util.concurrent.ThreadPoolExecutor$Worker.run          ThreadPoolExecutor.java:  617
 java.util.concurrent.ThreadPoolExecutor.runWorker          ThreadPoolExecutor.java: 1142
               java.util.concurrent.FutureTask.run                  FutureTask.java:  266
                                               ...                                       
               clojure.core/binding-conveyor-fn/fn                         core.clj: 1938
                                 boot.core/boot/fn                         core.clj: 1030
                                               ...                                       
                         boot.core/construct-tasks                         core.clj:  992
                                clojure.core/apply                         core.clj:  646
                                               ...                                       
                          boot.user/eval1467/fn/fn  boot.user642982767582347347.clj:   25
                                clojure.core/apply                         core.clj:  646
                                               ...                                       
                   degree9.boot-semver/eval1307/fn                  boot_semver.clj:   92
                                               ...                                       
               degree9.boot-semver.impl/eval985/fn                         impl.clj:   68
                                               ...                                       
                        boot.cli/separate-cli-opts                          cli.clj:  212
                                               ...                                       
            boot.from.clojure.tools.cli/parse-opts                          cli.clj:  499
                                               ...                                       
         boot.from.clojure.tools.cli/tokenize-args                          cli.clj:   27
                               clojure.core/re-seq                         core.clj: 4692
                           clojure.core/re-matcher                         core.clj: 4667
java.lang.ClassCastException: degree9.boot_semver.impl$version_impl$fn__1053 cannot be cast to java.lang.CharSequence
  clojure.lang.ExceptionInfo: degree9.boot_semver.impl$version_impl$fn__1053 cannot be cast to java.lang.CharSequence
    file: "/tmp/boot.user642982767582347347.clj"
    line: 33
```

**Comment by jgrzebyta** [comment]
*version-ns* & *version-file* in `impl.clj` should be functions not tasks.

**Comment by jgrzebyta** [comment]
fixed in version 1.6.3
